File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) 
    {
        int n = nums.size();
        
        if(nums.size()==1 )
        {
            return;
        }
        k=k%n;
        vector<int> lux;
        for(int i=n-k ; i<n;i++)
        {
            lux.push_back(nums[i]);
        }
        for(int i=0 ;i<n-k;i++)
        {
            lux.push_back(nums[i]);
        }
        nums=lux;

    }
    
};

File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k =k % n

        def spm(start:int , end:int )->None:
            while start<end:
                nums[start] , nums[end]=nums[end],nums[start]
                
                start+=1
                end-=1

        spm(0,n-1)
        spm(0,k-1)
        spm(k , n-1)        